sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -198,7 +198,7 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
 
         // Select the result of the first aggregate in the last aggregate.
         val result = AggregateExpression(
-          aggregate.First(evalWithinGroup(regularGroupId, operator.toAttribute), Literal(true)),
+          new aggregate.First(evalWithinGroup(regularGroupId, operator.toAttribute), Literal(true)),
```

Comment:
so adding more constructors doesn't help us to reduce diff, how about
```
case class First(funcName: String, child: Expression, ignoreNullsExpr: Expression) {
  def this(funcName: String, child: Expression)
}
object First {
  def apply(child: Expression, ignoreNullsExpr: Expression) ...
  def apply(child: Expression) ...
}
```

Reply:
Done